tests/test_main_check_connection.py

```
@@ -0,0 +1,618 @@
+        mock_matrix_client.add_event_callback = MagicMock()
+        mock_matrix_client.close = AsyncMock()
+        mock_connect_matrix.return_value = mock_matrix_client
+
+        mock_connect_meshtastic.return_value = MagicMock()
+
+        _reset_all_mmrelay_globals()
+        with pytest.raises(RuntimeError) as exc_info:
+            try:
+                asyncio.run(main(config))
+            finally:
+                _reset_all_mmrelay_globals()
```

Ruff (0.15.12)

[warning] `pytest.raises()` block should contain a single simple statement

(PT012)